src/cmap/test/mod.rs

```diff
@@ -128,129 +134,143 @@ impl Executor {
+                    let conn = state
+                        .connections
+                        .write()
+                        .await
+                        .remove(&connection)
+                        .unwrap();
+
+                    if let Some(pool) = state.pool.read().await.deref() {
+                        pool.check_in(conn).await;
+                    }
                 }
-            }
-            Operation::Clear => {
-                if let Some(pool) = state.pool.write().unwrap().deref() {
-                    pool.clear();
+                Operation::Clear => {
+                    if let Some(pool) = state.pool.write().await.deref() {
+                        pool.clear().await;
+                    }
                 }
+                Operation::Close => {
+                    state.connections.write().await.clear();
+                    state.pool.write().await.take();
+                }
+
+                // We replace all instances of `Start` with `StartHelper` when we preprocess the events,
+                // so this should never actually be found.
+                Operation::Start { .. } => unreachable!(),
             }
-            Operation::Close => {
-                state.connections.write().unwrap().clear();
-                state.pool.write().unwrap().take();
-            }
+            Ok(())
+        }.boxed()
+    }
+}
 
-            // We replace all instances of `Start` with `StartHelper` when we preprocess the events,
-            // so this should never actually be found.
-            Operation::Start { .. } => unreachable!(),
-        }
+impl Matchable for TlsOptions {
+    fn content_matches(&self, expected: &TlsOptions) -> bool {
+        self.allow_invalid_certificates.matches(&expected.allow_invalid_certificates) &&
+            self.ca_file_path.matches(&expected.ca_file_path) &&
+            self.cert_key_file_path.matches(&expected.cert_key_file_path)
+    }
+}
 
-        Ok(())
+impl Matchable for ConnectionPoolOptions {
+    fn content_matches(&self, expected: &ConnectionPoolOptions) -> bool {
+        self.app_name.matches(&expected.app_name) &&
+            self.connect_timeout.matches(&expected.connect_timeout) &&
+            self.credential.matches(&expected.credential) &&
+            self.max_idle_time.matches(&expected.max_idle_time) &&
+            self.max_pool_size.matches(&expected.max_pool_size) &&
+            self.min_pool_size.matches(&expected.min_pool_size) &&
+            self.tls_options.matches(&expected.tls_options) &&
+            self.wait_queue_timeout.matches(&expected.wait_queue_timeout)
     }
 }
 
-fn assert_events_match(actual: &Event, expected: &Event) {
-    match (actual, expected) {
-        (Event::ConnectionPoolCreated(actual), Event::ConnectionPoolCreated(ref expected)) => {
-            if let Some(ref expected_options) = expected.options {
-                assert_eq!(actual.options.as_ref(), Some(expected_options));
+impl Matchable for Event {
```

Comment:
since these tests use the "MATCH" function, I figured it was useful to use our test runner's facilities for that. This also helped catch a bug in the event emission.